services/supervisor/src/transactions.rs

```diff
@@ -131,6 +128,47 @@ impl StopService {
     }
 }
 
+impl ResumeService {
+    fn validate(&self, context: &CallContext<'_>) -> Result<(), ExecutionError> {
+        let instance = get_instance(context, self.instance_id)?;
+
+        match instance.status {
```

Comment:
Nit: you can just write `if instance.status != InstanceStatue::Stopped`

Reply:
Really ? `Option` derives `PartialOrd` and `Eq`. Take a look [here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=b3ff083e7147f89c341020d7392a1f31)

Reply:
You are right, today I am a little bit stupid.